coremltools/converters/mil/mil/passes/defs/optimize_normalization.py

```diff
@@ -214,8 +221,8 @@ def _try_apply_transform(
             x = mb.layer_norm(
                 x=x if is_require_rank4_transpose else reduce_op.x,
                 axes=axes,
-                gamma=gamma_var,
-                beta=beta_var,
+                gamma=gamma_var.val if is_new_const else gamma_var,
```

Comment:
By "materialize" `gamma and beta`,
this could potentially introduce more const.
Can you explain why are you using `.val` here?

Reply:
In the case where an elementwise affine is set in LayerNormANE, the existing gamma and beta var are the wrong shape: (1,C,1,1) but layer_norm MIL needs just (C).
Additionally, beta needs to be scaled by gamma to reverse the LayerNormANE scaling.
I assumed, but did not check, that the old gamma/beta consts would be pruned.

```python
gamma_var = mul_gamma_op.y if mul_gamma_op.x == add_beta_op.outputs[0] else mul_gamma_op.x
gamma_var = mb.const(
    val=np.squeeze(gamma_var.val),
    name="_fuse_layernorm_gamma",
)

# Scale beta by gamma. Note: this un-scaling introduces a little error.
# https://github.com/apple/ml-ane-transformers/blob/da64000fa56cc85b0859bc17cb16a3d753b8304a/ane_transformers/huggingface/distilbert.py#L31
beta_var = mb.const(
    val=np.squeeze(beta_var.val) * gamma_var.val,
    name="_fuse_layernorm_beta"
)
```

Is there a better way to handle either of those?

Separately, I am also creating 2 new consts for gamma and beta in the case where an elementwise affine was not set, but I think I can remove that since they are optional args in the layer_norm MIL op.

Reply:
Same as another comment:
https://github.com/apple/coremltools/pull/1972#discussion_r1349400607
No actions need to be done for now